antarest/study/storage/variantstudy/model/command/create_link.py

```diff
@@ -9,38 +9,76 @@
 # SPDX-License-Identifier: MPL-2.0
 #
 # This file is part of the Antares project.
-
+import typing as t
 from typing import Any, Dict, List, Optional, Tuple, Union, cast
 
-from pydantic import ValidationInfo, field_validator, model_validator
+from antares.study.version import StudyVersion
+from pydantic import Field, ValidationInfo, field_validator, model_validator
 
-from antarest.core.model import JSON
+from antarest.core.exceptions import LinkValidationError
+from antarest.core.serialization import AntaresBaseModel
+from antarest.core.utils.string import to_kebab_case
 from antarest.core.utils.utils import assert_this
 from antarest.matrixstore.model import MatrixData
 from antarest.study.model import STUDY_VERSION_8_2
+from antarest.study.storage.rawstudy.model.filesystem.config.links import AssetType, LinkStyle, TransmissionCapacity
 from antarest.study.storage.rawstudy.model.filesystem.config.model import FileStudyTreeConfig, Link
 from antarest.study.storage.rawstudy.model.filesystem.factory import FileStudy
 from antarest.study.storage.variantstudy.business.utils import strip_matrix_protocol, validate_matrix
 from antarest.study.storage.variantstudy.model.command.common import CommandName, CommandOutput, FilteringOptions
 from antarest.study.storage.variantstudy.model.command.icommand import MATCH_SIGNATURE_SEPARATOR, ICommand
 from antarest.study.storage.variantstudy.model.model import CommandDTO
 
+DEFAULT_COLOR = 112
+
 
-class LinkProperties:
-    HURDLES_COST: bool = False
-    LOOP_FLOW: bool = False
-    USE_PHASE_SHIFTER: bool = False
-    DISPLAY_COMMENTS: bool = True
-    TRANSMISSION_CAPACITIES: str = "enabled"
-    ASSET_TYPE: str = "ac"
-    LINK_STYLE: str = "plain"
-    LINK_WIDTH: int = 1
-    COLORR: int = 112
-    COLORG: int = 112
-    COLORB: int = 112
+class AreaInfo(AntaresBaseModel):
+    area1: str
+    area2: str
+
+
+class LinkInfoProperties(AntaresBaseModel):
+    hurdles_cost: bool = False
+    loop_flow: bool = False
+    use_phase_shifter: bool = False
+    transmission_capacities: TransmissionCapacity = TransmissionCapacity.ENABLED
+    asset_type: AssetType = AssetType.AC
+    display_comments: bool = True
+    colorr: int = Field(default=DEFAULT_COLOR, gt=0, lt=255)
+    colorb: int = Field(default=DEFAULT_COLOR, gt=0, lt=255)
+    colorg: int = Field(default=DEFAULT_COLOR, gt=0, lt=255)
+    link_width: float = 1
+    link_style: LinkStyle = LinkStyle.PLAIN
+
+    class Config:
+        alias_generator = to_kebab_case
+        populate_by_name = True
+
+
+class LinkInfoProperties820(LinkInfoProperties):
+    filter_synthesis: t.Optional[str] = None
+    filter_year_by_year: t.Optional[str] = None
```

Comment:
I think the best approach here, for both fields, would be to use an annotated type with a [custom serializer](https://docs.pydantic.dev/latest/concepts/serialization/#custom-serializers) and validation method.
This way, the fields can have their "real" type which is a `List[FilterOption]` while still being serialized and parsed as a `str`.

This will allow to re-use this annotated type in other places if needed.

Reply:
I added an annotated field that performs validation and serialization but the methods are not reusable